src/views/Migrate.vue

```diff
@@ -49,56 +214,56 @@ export default class Migrate extends Vue {
+
+        this.status = 'Retrieving your old Accounts...';
+        const legacyAccounts = await this.$rpc.keyguardClient.listLegacyAccounts();
 
-        if (!hasLegacyAccounts) {
-            this.title = 'Nothing to migrate.';
+        if (!legacyAccounts.length) {
+            this.deleteAccountsCache();
+            this.title = 'Nothing to update.';
             this.state = Loader.State.SUCCESS;
             setTimeout(() => this.$rpc.resolve([]), Loader.SUCCESS_REDIRECT_DELAY);
             return;
         }
 
-        this.status = 'Retrieving your legacy accounts...';
-        const legacyAccounts = await this.keyguardClient!.listLegacyAccounts();
-
-        if (!legacyAccounts.length) {
-            throw new Error('Could not get legacy accounts from Keyguard');
-        }
-
         this.status = 'Detecting vesting contracts...';
         const genesisVestingContracts = await (this.$refs.network as Network).getGenesisVestingContracts();
-        (this.$refs.network as Network).disconnect(); // Prevent syncing consensus
 
-        this.status = 'Storing your new accounts...';
+        this.status = 'Storing your new Accounts...';
         // For the wallet ID derivation to work, the ID derivation and storing of new wallets needs
         // to happen serially, e.g. synchroneous.
         const walletInfos: WalletInfo[] = [];
-        for (const account of legacyAccounts) {
-            const address = new Nimiq.Address(account.legacyAccount.address);
+        for (const keyInfo of legacyAccounts) {
+            const accountInfo = this.legacyKeyInfoObject2AccountInfo(keyInfo);
             const accounts = new Map<string, AccountInfo>([
-                [address.toUserFriendlyAddress(), new AccountInfo('m/0\'', account.legacyAccount.label, address)],
+                [accountInfo.userFriendlyAddress, accountInfo],
             ]);
 
-            const contracts = genesisVestingContracts.filter((contract) => contract.owner.equals(address));
+            const contracts = genesisVestingContracts.filter((contract) => contract.owner.equals(accountInfo.address));
 
             const walletInfo = new WalletInfo(
-                await WalletStore.deriveId(account.id),
-                account.id,
+                await WalletStore.deriveId(keyInfo.id),
+                keyInfo.id,
                 ACCOUNT_DEFAULT_LABEL_LEGACY,
                 accounts,
                 contracts,
                 WalletType.LEGACY,
+                /* keyMissing */ false,
+                /* fileExported */ false,
+                /* wordsExported */ true,
             );
 
             await WalletStore.Instance.put(walletInfo);
 
             walletInfos.push(walletInfo);
         }
 
-        this.status = 'Migrating Keyguard...';
-        await this.keyguardClient!.migrateAccountsToKeys();
+        this.status = 'Updating Keyguard...';
+        await this.$rpc.keyguardClient.migrateAccountsToKeys();
+        this.deleteAccountsCache();
```

Comment:
This means this info will not be available on `tryAgain()` anymore.
Is that ok?

Reply:
The `migrateAccountsToKeys()` action is the last action that can go wrong. If that succeeds and the code reaches this line, then nothing else can go wrong, everything is stored and the users gets the success screen.